Examples/Scripts/Python/full_chain_odd.py

```diff
@@ -116,17 +174,17 @@
 rnd = acts.examples.RandomNumbers(seed=42)
```

Comment:
We need to have this random number seed configurable from the command line as well. Just like it is for `sim_digi_odd.py`. Otherwise one can't really produce large samples. (Unless producing the large sample over ages, on a single thread...)

Reply:
note that you can also use skip to achieve this

Reply:
:confused: Only if I want to generate the same events N times.

Right now, since the script sets a very explicit random seed (which is good...), it should always produce the same events. Unlike `sim_digi_odd.py`, which does have a way of setting the seed.
  - https://github.com/acts-project/acts/blob/main/Examples/Scripts/Python/sim_digi_odd.py#L72-L77
  - https://github.com/acts-project/acts/blob/main/Examples/Scripts/Python/sim_digi_odd.py#L95

That, in combination with `--skip` is how one can do large scale simulations... :thinking:

Reply:
Ok, for the moment you'll have to do that locally, will update the PR accordingly, but not before Monday

Reply:
What I did in the past is using the same seed but then split the workload across multiple processes using `--skip N --events M`.

```
full_chain_odd.py -n 1000 --skip 0 &
full_chain_odd.py -n 1000 --skip 1000 &
...
```

something like that and afterwards I stacked the root files from the different output folders into one big file.

That does not work for you in this case?

Reply:
> For AndiW: But what you describe would produce the same events, just in different files / with different event numbers.

It depends on the design. If the for loop is designed smartly enough so that the random generator is called for the skipped event, then we can use AndiW's suggestion. That's what I did in the detray validation test

The drawback of @krasznaa's method is that it is difficult for other people to reproduce the same data with different configuration of number of threads because it will have different seeds

Reply:
In our Examples Framework the seed is actually offset per event. The event seed is basically `seed + event_nr` so using the same seed you should aways get the same random numbers for a specific event number no matter how you arrive at this event number.

```
full_chain_odd.py -n2
```

and

```
full_chain_odd.py -n1
full_chain_odd.py -n1 -s1
merge
```

should give the same result basically

Reply:
Of course, supporting seed configuration is always a good idea

Reply:
Number of threads? The random seed -> job assignment is a single-threaded affair in the Python script I point to. As I wrote earlier, one could add

```python
random.seed(42)
```

at the beginning of that script, and then, as long as Acts actually respects the seed that it is given, all hundreds of simulation jobs should produce the exact same output (modulo floating point differences) on all runs, on all machines.

But since reproducibility is not a concern of mine at the moment (especially since I can't generate as many ttbar events at once as I'd like), getting different simulations on every run is actually what I'm after. (I had >1 TB space for this. And I ran out...)

Reply:
> In our Examples Framework the seed is actually offset per event. The event seed is basically `seed + event_nr` so using the same seed you should aways get the same random numbers for a specific event number no matter how you arrive at this event number.
> 
> ```
> full_chain_odd.py -n2
> ```
> 
> and
> 
> ```
> full_chain_odd.py -n1
> full_chain_odd.py -n1 -s1
> merge
> ```
> 
> should give the same result basically

In the end, Acts is not about simulation per se. So I guess that confusing setup is okay... I mean, re-seeding the random number generator event after event is a good thing per se. But it should be better documented then. :thinking: (And it still doesn't change it that the user should be allowed to change the initial seed nevertheless.)

Reply:
I understood that you want to run the scripts with multiple threads with different seed. Sorry if that is not the case

Reply:
> getting different simulations on every run is actually what I'm after. 

Is this non-deterministic simulation normal in ATLAS simulation? I thought it is more reasonable to provide a sequence of random numbers to the seed by users not by the program, only if you need to do.

Reply:
I think for a proper random number stream implementation we would have to invest quite some work, I am not sure if this is worth it for a showcase model.